### PR TITLE
Stream output

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [ ] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added new automated tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the section below.

### Changes description
...

### CODE REVIEW

File: src/snowcli/output/printing.py [hunk 1]
```diff
@@ -25,99 +28,89 @@ def default(self, o):
...
+    def get_data(self) -> Iterator[Union[Dict, OutputData]]:
+        """Returns iterator over output data"""
+        if not self._stream:
+            return None
 
+        for item in self._stream:
+            if isinstance(item, OutputData):
+                yield item
+            else:
+                yield item
```
Comment (sfc-gh-astus) [inline src/snowcli/output/printing.py:94]: if return item else return item? 🤔

File: src/snowcli/output/printing.py [hunk 2]
```diff
@@ -155,40 +144,21 @@ def _get_format_type() -> OutputFormat:
...
+    for column in item.keys():
         table.add_column(column)
-    for row in data:
-        table.add_row(*[str(i) for i in row.values()])
-    print(table)
+    with Live(table, refresh_per_second=4):
+        table.add_row(*[str(i) for i in item.values()])
+        for item in stream:
+            table.add_row(*[str(i) for i in item.values()])
+    print("")
```
Comment (sfc-gh-astus) [inline src/snowcli/output/printing.py:164]: This print can be removed?

File: tests/output/test_printing.py [hunk 3]
```diff
@@ -147,25 +214,25 @@ def test_print_different_data_sources_json(capsys):
...
 
 def test_print_with_no_data_json(capsys):
-    output_data = OutputData(format=OutputFormat.JSON)
+    output_data = OutputData(format_=OutputFormat.JSON)
 
-    print_output(output_data)
+    output_data.print()
 
-    assert _get_output(capsys) == "Done\n"
+    assert _get_output(capsys) == "No data\n"
```
Comment (sfc-gh-astus) [inline tests/output/test_printing.py:235]: Maybe _get_output should return string after strip()?

File: src/snowcli/output/printing.py (hunk 1 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/output/printing.py:94]: Good catch 😄

File: src/snowcli/output/printing.py (hunk 2 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/output/printing.py:164]: Let me check, it had some purpose to be aligned with previous tests 😄

File: tests/output/test_printing.py [hunk 4]
```diff
@@ -46,15 +48,74 @@ def test_print_multi_cursors_table(capsys):
...
 
 
-def test_print_different_data_sources_table(capsys):
-    output_data = (
-        OutputData(format=OutputFormat.TABLE)
-        .add_cursor(_create_mock_cursor())
-        .add_string("Command done")
-        .add_list([{"key": "value"}])
+def test_print_different_multi_cursors_table(capsys, mock_cursor):
+    output_data = OutputData().from_list(
```
[the remark this answers is not in the document]
Comment (sfc-gh-turbaszek) [reply tests/output/test_printing.py:52]: Both work but version without `()` is correct

File: src/snowcli/output/printing.py [hunk 5]
```diff
@@ -25,99 +28,85 @@ def default(self, o):
...
-                else:
-                    yield {"result": output}
-            elif isinstance(output, list):
-                yield output
-
-    def size(self) -> int:
-        return self.counter
-
+    def empty(self):
+        return self._stream is None
```
Comment (sfc-gh-astus) [inline src/snowcli/output/printing.py:83]: What with case when stream has 0 items?

Comment (sfc-gh-turbaszek) [reply src/snowcli/output/printing.py:83]: Can stream have 0 items and not be None?

Comment (sfc-gh-astus) [reply src/snowcli/output/printing.py:83]: You are the python expert 😄

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**